backend/user/serializers.py

```diff
@@ -11,14 +11,14 @@ class Meta:
 
     def create(self, validated_data):
         validated_data["user"] = self.context["request"].user
-        token = NotificationToken.objects.filter(user=validated_data["user"]).first()
+        token_obj = NotificationToken.objects.filter(user=validated_data["user"]).first()
 
-        if token:
-            # if token already exists, just update it
-            token.kind = validated_data["kind"]
-            token.token = validated_data["token"]
-            token.save()
-            return token
+        if token_obj:
```

Comment:
Hm, now that I am looking at this, I think I messed up last semester lol. The code here is basically just redefining the `update(...)` function inherited from the serializer. What I am thinking of now:

1. Removing this code, and overriding the `update(self, instance, validated_data)` function to just error if they try to update the user, and the call the `super().update(...)` function. This update function will handle all PATCH requests. Example of overriding [here](https://github.com/pennlabs/penn-mobile/blob/master/backend/portal/serializers.py#L107)
2. Erroring out in the `create(...)` function whenever there already exists a token for the user